src/nvim/path.c

```diff
@@ -360,7 +360,7 @@ void add_pathsep(char *p)
   FUNC_ATTR_NONNULL_ALL
 {
   if (*p != NUL && !after_pathsep(p, p + strlen(p)))
-    strcat(p, PATHSEPSTR);
+    STRLCAT(p, PATHSEPSTR, MAXPATHL);
```

Comment:
And, by the way, I would write it as

```C
const size_t len = strlen(p);
if (*p != NUL && !after_pathsep(p, p + len) && len < MAXPATHL - 1) {
  memcpy(p + len, PATHSEPSTR, sizeof(PATHSEPSTR));
}
```

: there is no need to compute length twice. But this is only optimization, no other reasons.